pkgs/build-support/vm/linuxkit-builder/default.nix

```diff
@@ -0,0 +1,289 @@
+
+    mkdir -p /etc/acpi/PWRF /etc/acpi/events
+    cat >/etc/acpi/PWRF/00000080 <<EOF
+    #!/bin/sh
+    poweroff
+    EOF
+    chmod +x "$pkgdir"/etc/acpi/PWRF/00000080
+    # ${pkgsLinux.acpid}/bin/acpid -d
+
+    ${pkgsLinux.openssh}/bin/sshd -e -f ${sshdConfig}
+
+    ${pkgsLinux.go-vpnkit}/bin/vpnkit-forwarder &
+    ${pkgsLinux.go-vpnkit}/bin/vpnkit-expose-port \
+      -i \
+      -host-ip 127.0.0.1 -host-port ${hostPort} \
+      -container-ip 192.168.65.2 -container-port 22 \
+      -no-local-ip &
+
+    mkdir -p /dev/input /etc/acpi/PWRF /var/log
+    mknod /dev/input/event0 c 13 64
+    cat <<EOF > /etc/acpi/PWRF/00000080
+    #!/bin/sh
+    poweroff -f
+    EOF
+    chmod +x /etc/acpi/PWRF/00000080
+
+    echo
+    echo -e '\033[91;47m======================================================================'
+    echo 'Remote builder has started.'
+    echo
+    echo 'If this is a fresh VM you need to run the following on the host:'
+    echo '    sudo ~/.nixpkgs/linuxkit-builder/root-init.sh'
+    echo
+    echo 'Then you can use the following to build x86_64-linux derivations:'
+    echo '    sudo $(cat ~/.nixpkgs/linuxkit-builder/env) nix-build'
+    echo
+    echo 'Exit this VM by running:'
+    echo '    kill $(cat ~/.nixpkgs/linuxkit-builder/nix-state/hyperkit.pid)'
+    echo '======================================================================'
+    echo -en '\033[0m'
+
+    exec ${pkgsLinux.busybox}/bin/acpid -f
+  '';
+
+  img = "bzImage";
+  initrd = makeInitrd {
+    contents = [
+      { object = stage1Init;
+        symlink = "/init";
+      }
+    ];
+  };
+  startLinuxKitBuilder = ''
+    #!${bash}/bin/bash
+
+    usage() {
+      echo "Usage: $(basename $0) [-d directory] [-f features] [-s size] [-c cpus] [-m mem]" >&2
+    }
+
+    NAME="linuxkit-builder"
+
+    DIR="$HOME/.nixpkgs/$NAME"
+    FEATURES="big-parallel"
+    SIZE="10G"
+    CPUS=1
+    MEM=1024
+    while getopts "d:f:s:h" opt; do
+      case $opt in
+        d) DIR="$OPTARG" ;;
+        f) FEATURES="$OPTARG" ;;
+        s) SIZE="$OPTARG" ;;
+        c) CPUS="$OPTARG" ;;
+        m) MEM="$OPTARG" ;;
+        h | \?)
+          usage
+          exit 64
+          ;;
+      esac
+    done
+
+    mkdir -p $DIR
+    ln -fs ${linuxkitKernel}/${img} $DIR/nix-kernel
+    ln -fs ${initrd}/initrd $DIR/nix-initrd.img
+    echo -n "console=ttyS0 panic=1 command=${stage2Init} loglevel=7 debug" > $DIR/nix-cmdline
+
+    [ ! -f $DIR/key ] && {
+      cp '${privateKey}' $DIR/key
+      chmod 0400 $DIR/key
+    }
+    echo "root@localhost x86_64-linux $DIR/key $CPUS 1 $FEATURES" > $DIR/remote-systems
+
+    cat <<EOF > $DIR/root-init.sh
+    #!/usr/bin/env bash
+    if [[ \$EUID -ne 0 ]]; then
+      echo "This script should be executed as root." 1>&2
+      exit 1
+    fi
+    chown root $HOME/.nixpkgs/linuxkit-builder/key
+    ssh-keygen -R '[localhost]:${hostPort}'
+    ssh-keyscan -t ecdsa-sha2-nistp256 -p ${hostPort} localhost >> /var/root/.ssh/known_hosts
```

Comment:
On nixos this returns nothing:
```
$ ssh-keyscan -t ecdsa-sha2-nistp256 localhost
# localhost:22 SSH-2.0-OpenSSH_7.5
```

while 

```console
$ ssh-keyscan -t ed25519 localhost
# localhost:22 SSH-2.0-OpenSSH_7.5
localhost ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDIWYJWReBSOb7XZ+taCH7UlshY8Fysubm9PNGzAiGS/8NpRSZcdLaWxroJ0dF6EpSw1o8eeO6by3DkduYxVFnQwwhGvOmYfaGTqYDxSBnBr0L9hgOBSd4HEIdDgs9pFoWVQaXMy+lueV5AiK8dtmtwXs8C8O/hvz80P15MtMCYRu1UhJW3u40zR5TWbVxXDNkAd3kBvuTRznd1fVJTJAp2S1/yR8Rhx15x+w0ty7X9he5C99kAcPJi5ofe/M2xWkWB66iprvljSSl7wEVSOSKbl0uFbsy8d7utLilIAN4yBQ01aErJkpTpteRakSU6G0D+MR1NPqXIoieu4weKcQ7aXbnBe8YOOxWooAdjQ6oSqqyWvAOI3I69N0rlelzcm800w14pk6c+K4bKq+Ds/NHYbu6HEkTluvhtP3aRkEtPBis6H+wV6lREXI9ko3uCi+Bydyg+pDl8ti0LWKRECHKTJ9fu7pabE+CSBSRrNUFuC43UQWMjLHs1R1ec6TW1+McyOvOxaQF4KoBnLYzRMcq+muno6kS66R/oJnpflKH7xrG0wnk1hxGmhiMroExTINMcwt7EzvON4rwT1Jwmsxm2M+0Kpj0iGCQIQ34z1d5nwK3QOXD9hbqdqkFFPn7fBZgHhjWOotk9evxq9nHKh+xS50mqBoyrmv+xsE/txRbemQ==
# localhost:22 SSH-2.0-OpenSSH_7.5
# localhost:22 SSH-2.0-OpenSSH_7.5
localhost ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIM0sA09EdzAF25+oVGyMamLmsXrl4oIx6/endHqZ9lBa
$ ssh-keyscan -t ed25519 localhost
# localhost:22 SSH-2.0-OpenSSH_7.5
localhost ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIM0sA09EdzAF25+oVGyMamLmsXrl4oIx6/endHqZ9lBa
```

So apple is still on nist cyphers by default here?
```

Reply:
ecdsa-sha2-nistp256 seems to be the default on macOS. I'm happy to leave out the `-t` altogether though.